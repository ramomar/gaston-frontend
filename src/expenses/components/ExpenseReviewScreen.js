import React from 'react';
import PropTypes from 'prop-types';
import Shapes from '../shapes';
import { Text, Button } from 'grommet';
import { LinkPrevious } from 'grommet-icons';
import { Screen, ScreenHeader, ScreenBody } from '../../foundation/components/screen';
import ExpenseReviewForm from './ExpenseReviewForm';
import { DateTime } from 'luxon';
import * as R from 'ramda';

function ExpenseReviewScreen(props) {
  const date = DateTime.fromISO(props.expense.date);

  const expenseWithLuxonDate = R.assoc('date', date, props.expense);

  const formattedDay = date.toFormat(`d 'de' MMMM`);

  return (
    <Screen>
      <ScreenHeader
        start={<Button plain icon={<LinkPrevious />} onClick={props.goToExpenses} />}
        center={<Text weight='bold' size='large'>{`Gasto del ${formattedDay}`}</Text>}
      />
      <ScreenBody>
        <ExpenseReviewForm
          expense={expenseWithLuxonDate}
          reviewExpense={props.reviewExpense}
          expenseCategories={props.expenseCategories}
        />
      </ScreenBody>
    </Screen>
  );
}

ExpenseReviewScreen.propTypes = {
  expense: PropTypes.shape({
    id: PropTypes.string.isRequired,
    note: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    date: PropTypes.string.isRequired
  }).isRequired,
  goToExpenses: PropTypes.func.isRequired,
  reviewExpense: PropTypes.func.isRequired,
  expenseCategories: PropTypes.arrayOf(Shapes.expenseCategory).isRequired
};

export default ExpenseReviewScreen;
