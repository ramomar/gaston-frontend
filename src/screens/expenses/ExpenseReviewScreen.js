import React from 'react';
import { useLocation, useHistory } from 'react-router-dom';
import { Text, Button } from 'grommet';
import { LinkPrevious } from 'grommet-icons';
import { Screen, ScreenHeader } from '../components';
import ExpenseReviewForm from './components/ExpenseReviewForm';
import { DateTime } from 'luxon';
import { assoc } from 'ramda';

function ExpenseReviewScreen(props) {
  const { state: { expense } } = useLocation();
  const { goBack } = useHistory();

  // See ExpenseListItem
  const date = DateTime.fromISO(expense.date);
  const expenseWithLuxonDate = assoc('date', date, expense);

  const formattedDay = date.toFormat(`d 'de' MMMM`);

  return (
    <Screen>
      <ScreenHeader
        start={<Button plain icon={<LinkPrevious />} onClick={goBack} />}
        center={<Text weight='bold' size='large'>{`Gasto el ${formattedDay}`}</Text>}
      />
      <ExpenseReviewForm expense={expenseWithLuxonDate} />
    </Screen>
  );
}

export default ExpenseReviewScreen;
