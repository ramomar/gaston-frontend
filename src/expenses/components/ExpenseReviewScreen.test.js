import React from 'react';
import renderer from 'react-test-renderer';
import { Grommet } from 'grommet';
import ExpenseReviewScreen from './ExpenseReviewScreen';
import { DateTime } from 'luxon';

it('renders correctly', () => {
  const now = DateTime.fromISO('2019-12-29T00:00:00.000Z');

  const dateTimeLocalMock = jest.fn();

  dateTimeLocalMock.mockReturnValue(now);

  DateTime.local = dateTimeLocalMock;

  const expense = {
    'id': '0007182d-54cb-42b7-88fc-bbaba51db198',
    'amount': 150,
    'date': DateTime.fromISO('2017-03-19T05:29:02.700Z'),
    'note': 'Cena'
  };

  const expenseCategories = [
    { name: 'Educaci√≥n' },
    { name: 'Cena' }
  ];

  const component =
    <Grommet>
      <ExpenseReviewScreen
        expense={expense}
        goToExpenses={() => null}
        reviewExpense={() => null}
        expenseCategories={expenseCategories}>
      </ExpenseReviewScreen>
    </Grommet>;

  const tree = renderer.create(component)

  expect(tree).toMatchSnapshot();
});
