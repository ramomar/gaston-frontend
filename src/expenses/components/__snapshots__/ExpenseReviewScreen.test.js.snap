// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <div
    className="StyledBox-sc-13pk1d4-0 hCTAHO"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 fpgWwd"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 koruWX"
      >
        <button
          className="StyledButton-sc-323bzc-0 laBEqz"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <svg
            aria-label="LinkPrevious"
            className="StyledIcon-ofa7kd-0 iOkQrb"
            viewBox="0 0 24 24"
          >
            <path
              d="M2,12 L22,12 M13,3 L22,12 L13,21"
              fill="none"
              stroke="#000"
              strokeWidth="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </button>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 koruWX"
      >
        <span
          className="StyledText-sc-1sadyjn-0 dFefMJ"
          size="large"
        >
          Gasto del 18 de March
        </span>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 koruWX"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 hCTAHO"
    >
      <form
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 dVeKXC FormField__FormFieldBox-m9hood-0 eshLKp"
        >
          <label
            className="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            Nota
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gwljPq"
          >
            <textarea
              autoFocus={true}
              className="StyledTextArea-sc-17i3mwp-0 gIdccB"
              label="Nota"
              name="note"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Cena del viernes en la noche"
              value="Cena"
            />
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dVeKXC FormField__FormFieldBox-m9hood-0 eshLKp"
        >
          <label
            className="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            Categoría
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gwljPq"
          >
            <button
              aria-label="Open Drop"
              className="StyledButton-sc-323bzc-0 jfbKsv Select__StyledSelectDropButton-sc-17idtfo-1 cvpccN"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 cGyFlY"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 gbHXHp"
                >
                  <div
                    className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
                  >
                    <input
                      autoComplete="off"
                      className="StyledTextInput-sc-1x30a0s-0 duvNfq Select__SelectTextInput-sc-17idtfo-0 lmyIUy"
                      label="Categoría"
                      name="category"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Educación"
                      readOnly={true}
                      tabIndex="-1"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="StyledBox-sc-13pk1d4-0 lgSzcm"
                  style={
                    Object {
                      "minWidth": "auto",
                    }
                  }
                >
                  <svg
                    aria-label="FormDown"
                    className="StyledIcon-ofa7kd-0 dSllCE"
                    viewBox="0 0 24 24"
                  >
                    <polyline
                      fill="none"
                      points="18 9 12 15 6 9"
                      stroke="#000"
                      strokeWidth="2"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dVeKXC FormField__FormFieldBox-m9hood-0 eshLKp"
        >
          <label
            className="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            Fecha
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gwljPq"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 kedzFm"
                label="Fecha"
                name="day"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="12/28/2019"
                value="3/18/2017"
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dVeKXC FormField__FormFieldBox-m9hood-0 eshLKp"
        >
          <label
            className="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            Hora
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gwljPq"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 kedzFm"
                label="Hora"
                name="hour"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="18:00"
                value="23:29"
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dVeKXC FormField__FormFieldBox-m9hood-0 eshLKp"
        >
          <label
            className="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            Cantidad
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gwljPq"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 kedzFm"
                label="Cantidad"
                name="amount"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="$ 250.00"
                value="$ 150.00"
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 NxMgH"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 koruWX"
        >
          <button
            className="StyledButton-sc-323bzc-0 cSMahv"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Terminar
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
